### ISSUE TITLE
Configuration for TLS termination and transparent proxy

### ISSUE THREAD

**Issue by tbarbette** [nokia/ClickNF#3]
Hi all!

Thanks for this very great paper/stack!

Is there an example configuration for the TLS termination? I'm not sure how SSLClient and SSLServer should be piped together.

Also, do you have an example for a transparent proxy, eg to implement an IDS that does something on the HTTP payload. I'm not sure how to "reduce" the socks4proxy to catch all flows, and emit a connection to the original destination (instead of the socks protocol). From there I guess I'll be able to run the pattern matcher on the packets going through by myself.

Thanks!
Tom

**Comment by gallomassimo** [comment]
Hi Tom,

For the TLS termination, I just added two configurations: echo-ssl-client-modular.click and echo-ssl-server-modular.click that uses TLS termination at Client and server side (the application is the echo client/server one). I did not tested a lot, especially with the ssl client side, so there might be issues there. 

For the transparent proxy, unfortunately we do not have anything at the moment. 

Hope that this can help you.
Max
